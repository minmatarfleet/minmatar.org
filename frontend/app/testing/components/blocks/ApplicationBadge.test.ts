import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import ApplicationBadge from "@components/blocks/ApplicationBadge.astro";

test("ApplicationBadge defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(ApplicationBadge, {
    props: {
      character: {
        character_id: 42,
        character_name: "Anni Todako",
        corporation: 1337,
      },
      corporation: {
        id: 1337,
        name: "chicken farmers",
      },
      type: {},
    },
  });

  expect(result).toMatchSnapshot();
});
