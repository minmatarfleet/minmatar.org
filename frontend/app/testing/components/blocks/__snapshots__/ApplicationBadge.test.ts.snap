// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ApplicationBadge defaults 1`] = `
"<div class="flex-block gap---space-2xs" data-astro-cid-dyfytgei="true" data-mod style="--flex_block_gap: var(--space-2xs);">  <div class="fixed-fluid pilot-item [ w-full ]" data-astro-cid-dyfytgei="true" data-astro-cid-xf65qrs5="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture data-astro-cid-hwwa7zih> <img loading="lazy" height="64" width="64" src="https://images.evetech.net/characters/42/portrait?size=64" x-bind:src="false ? get_player_icon(false) : 'https://images.evetech.net/characters/42/portrait?size=64'" alt="Profile picture of Anni Todako" x-bind:alt="\`Profile picture of \${(false ? false : &#34;Anni Todako&#34;)}\`" x-data="{
            show_character_modal() {
                if (!false &#38;&#38; 42 == 0)
                    return

                show_modal({
                    partial: '/partials/character_modal_component/?character_id='+(false ? false : '42'),
                })
            }
        }" x-on:click.stop.prevent="show_character_modal()" data-astro-cid-hwwa7zih> </picture>  <div class="!flex !items-center" data-astro-cid-xf65qrs5="true">  <div class="flex-block [ w-full ] gap-0" data-astro-cid-xf65qrs5="true" data-mod style="--flex_block_gap: 0;">  <span class="truncate text-[var(--highlight)] cursor-pointer max-w-fit" x-data="{
                    show_character_modal() {
                        if (42 == 0)
                            return

                        show_modal({
                            partial: '/partials/character_modal_component/?character_id=42',
                        })
                    }
                }" x-on:click.stop.prevent="show_character_modal()" data-astro-cid-xf65qrs5>Anni Todako</span>   <div class="fixed-fluid [ items-center ]" data-astro-cid-xf65qrs5="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-3xs);--fixed_fluid_width: 24px;--fixed_fluid_breakpoint: 50%;">  <picture data-astro-cid-xf65qrs5> <img loading="lazy" src="https://images.evetech.net/corporations/undefined/logo?size=32" width="24" height="24" alt="Profile picture of Anni Todako" data-astro-cid-xf65qrs5> </picture> <span class="[ truncate ]" data-astro-cid-xf65qrs5> <small data-astro-cid-xf65qrs5></small> </span>  </div>     </div>   </div>  </div>    <div class="wrapper !flex" data-astro-cid-dyfytgei="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: 23px;--wrapper_padding_inline_end: 23px;--wrapper_padding_block: 0;--wrapper_max_width: 64px;"> <svg width="18" height="20" viewBox="0 0 18 20" fill="none" xmlns="http://www.w3.org/2000/svg"> <path d="M7.80039 0.399902L10.2004 0.399902L10.2004 14.7999L15.0004 9.9999L17.4004 11.1999L9.00039 19.5999L0.600391 11.1999L3.00039 9.9999L7.80039 14.7999L7.80039 0.399902Z" fill="currentColor"></path> </svg> </div>   <div class="fixed-fluid [ items-center ]" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture class="border"> <img loading="lazy" src="https://images.evetech.net/corporations/1337/logo?size=64" width="64" height="64"> </picture> <div class="flex-block [ w-full ] gap-0" data-mod style="--flex_block_gap: 0;"> <span> <span class="[ text-[var(--highlight)] max-w-fit ]">chicken farmers</span> </span>
             <small></small>  </div>   </div>    </div>  "
`;
